airbyte-integrations/connectors/source-avni/source_avni/source.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import boto3
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import IncrementalMixin, Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.models import SyncMode
+
+
+class Avni(HttpStream, ABC):
+
+    url_base = "https://app.avniproject.org/api/"
+    primary_key = "ID"
+    cursor_value = None
+    current_page = 0
+    last_record = None
+    
+    def __init__(self, start_date: str, path , auth_token: str, **kwargs):
+        super().__init__(**kwargs)
+
+        self.start_date = start_date
+        self.auth_token = auth_token
+        self.stream=path
+
+
+class AvniStream(Avni,IncrementalMixin):
+
+    """
+    
+    This implement diffrent Stream in Source Avni
+    
+    Api docs : https://avni.readme.io/docs/api-guide
+    Api endpoints : https://app.swaggerhub.com/apis-docs/samanvay/avni-external/1.0.0
+    """
+    def path(self, **kwargs) -> str:
+        return self.stream
+    
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+
+        params = {"lastModifiedDateTime": self.state["Last modified at"]}
+        if next_page_token:
+            params.update(next_page_token)
+        return params
+    
+    @property
+    def name(self) -> str:
+        return self.stream
+    
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+
+        return {"auth-token": self.auth_token}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+
+        data = response.json()["content"]
+        if data:
+            self.last_record = data[-1]
+
+        yield from data
+
+    def update_state(self) -> None:
+
+        if self.last_record:
+            updated_last_date = self.last_record["audit"]["Last modified at"]
+            if updated_last_date>self.state[self.cursor_field[1]]:
+                self.state = {self.cursor_field[1]: updated_last_date}
+        self.last_record = None
+        return None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+
+        total_elements = int(response.json()["totalElements"])
+        page_size = int(response.json()["pageSize"])
+
+        if total_elements == page_size:
+            self.current_page = self.current_page + 1
+            return {"page": self.current_page}
+
+        self.update_state()
+
+        self.current_page = 0
+
+        return None
+
+    state_checkpoint_interval = None
+
+    @property
+    def cursor_field(self) -> List[str]:
+        return ["audit", "Last modified at"]
```

Comment:
Normalization can't handle list of cursors. You can transform the record to extract this info and keep in the first level of the object.